pkg/smartcontract/rpcbinding/binding.go

```diff
@@ -519,6 +553,30 @@ func dropStdEvents(events []manifest.Event, std *standard.Standard) []manifest.E
 	return events
 }
 
+// dropNep24Types removes NamedTypes of NEP-24 from the config if they are used only once.
+func dropNep24Types(cfg binding.Config) binding.Config {
+	var targetTypeName string
+	if royaltyInfo, ok := cfg.Types["royaltyInfo"]; ok && royaltyInfo.Value != nil {
```

Comment:
It won't work with the following case:
```
type RoyaltyRecipients []RoyaltyRecipient

// RoyaltyInfo returns a list of royalty recipients and the corresponding royalty amounts.
func RoyaltyInfo(tokenID []byte, royaltyToken interop.Hash160, salePrice int) RoyaltyRecipients {
```

Add a comment describing what you're looking for under this `if`. And this case is also not hard to fix.

Reply:
Let me check it.

Reply:
Also added type checks of returning value.

Reply:
Good, it works, I just thought that type aliases should be handled separately.